packages/common/src/ui/layout/tables/utils/ColumnSetBuilder.ts

```diff
@@ -0,0 +1,161 @@
+import { getCheckboxSelectionColumn } from './../columns/CheckboxSelectionColumn';
+import { ColumnAlign } from '@openmsupply-client/common/src/ui/layout/tables/columns/types';
+import { ColumnFormat } from './../columns/types';
+import { DomainObject } from './../../../../types';
+import { ColumnDefinition } from '../columns/types';
+
+const createColumn = <T extends DomainObject>(
+  column: ColumnDefinition<T>
+): ColumnDefinition<T> => {
+  return column;
+};
+
+export type ColumnKey =
+  | 'invoiceNumber'
+  | 'type'
+  | 'status'
+  | 'entered'
+  | 'confirmed'
+  | 'total'
+  | 'comment'
+  | 'selection'
+  | 'name'
+  | 'code'
+  | 'packSize'
+  | 'quantity';
+
+const getColumnLookup = <T extends DomainObject>(): Record<
+  ColumnKey,
+  ColumnDefinition<T>
+> => ({
+  name: {
+    key: 'name',
+    label: 'label.name',
+    width: 75,
+  },
+  invoiceNumber: {
+    key: 'invoiceNumber',
+    label: 'label.invoice-number',
+    width: 75,
+  },
+  type: {
+    label: 'label.type',
+    key: 'type',
+    width: 150,
+  },
+  status: {
+    label: 'label.status',
+    key: 'status',
+    width: 100,
+  },
+  entered: {
+    label: 'label.entered',
+    key: 'entered',
+    format: ColumnFormat.date,
+    width: 100,
+  },
+  confirmed: {
+    label: 'label.confirmed',
+    key: 'confirmed',
+    format: ColumnFormat.date,
+    width: 100,
+  },
+
+  total: {
+    label: 'label.total',
+    key: 'total',
+    width: 75,
+    align: ColumnAlign.Right,
+  },
+  comment: {
+    label: 'label.comment',
+    key: 'comment',
+    width: 250,
+  },
+  selection: getCheckboxSelectionColumn(),
+  code: {
+    label: 'label.code',
+    key: 'code',
+    width: 20,
+  },
+  packSize: {
+    label: 'label.packSize',
+    key: 'packSize',
+    width: 20,
+    align: ColumnAlign.Right,
+  },
+  quantity: {
+    label: 'label.quantity',
+    key: 'quantity',
+    width: 20,
+    align: ColumnAlign.Right,
+  },
+});
+
+export class ColumnSetBuilder<T extends DomainObject> {
```

Comment:
In general I think this is OK, not great. 

I'm still not fully sure how we'd handle customised columns yet but I think this gets us close..